server/utils/mpesa_utils.py

```
@@ -0,0 +1,278 @@
+            expires_in = token_data.get('expires_in', 3600)
+            try:
+                expires_in = int(expires_in)
+            except (ValueError, TypeError):
+                expires_in = 3600
+
+            # Set expiry time with safety margin
+            self.expiry_time = datetime.now(timezone.utc).timestamp() + expires_in - 60
+            return self.token
+
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"Failed to get MPESA token: {str(e)}")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)